### PR TITLE
MGMT-8755: Make openshift_version optional for day2 cluster in REST API

### PR DESCRIPTION
**Pull request by avishayt**
# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -802,27 +802,14 @@ func (b *bareMetalInventory) V2ImportClusterInternal(ctx context.Context, kubeKe
 	log := logutil.FromContext(ctx, b.log).WithField(ctxparams.ClusterId, id)
 	apivipDnsname := swag.StringValue(params.NewImportClusterParams.APIVipDnsname)
 	clusterName := swag.StringValue(params.NewImportClusterParams.Name)
-	inputOpenshiftVersion := swag.StringValue(params.NewImportClusterParams.OpenshiftVersion)
+	inputOpenshiftVersion := params.NewImportClusterParams.OpenshiftVersion
```
Comment (rollandf) [inline internal/bminventory/inventory.go:805]: so what is `inputOpenshiftVersion` used for now?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -848,7 +833,24 @@ func (b *bareMetalInventory) V2ImportClusterInternal(ctx context.Context, kubeKe
 		KubeKeyNamespace: kubeKey.Namespace,
 	}
 
-	err = validations.ValidateClusterNameFormat(clusterName)
+	// Day2 supports only x86_64 for now
+	if inputOpenshiftVersion != "" {
```
Comment (eranco74) [inline internal/bminventory/inventory.go:837]: I'm not sure why we need to bother about all this?
What code in the add host flow require an openshift release image?

Comment (eranco74) [reply internal/bminventory/inventory.go:837]: Is it just for validating the assisted-service "knows" this release?
it makes sense for the day1 cluster where the OCP release is being used but in day 2 I don't think the assisted service has any use for it so I think the assisted service can be version agnostic.

File: internal/host/hostcommands/install_cmd.go [hunk 3]
```diff
@@ -142,12 +121,35 @@ func (i *installCmd) getFullInstallerCommand(cluster *common.Cluster, host *mode
...
+		if err != nil {
+			return "", err
+		}
+		mustGatherImages, err := i.getMustGatherArgument(mustGatherMap)
+		if err != nil {
+			return "", err
+		}
+
+		installerCmdArgs = append(installerCmdArgs, "--must-gather-image", mustGatherImages)
+		installerCmdArgs = append(installerCmdArgs, "--openshift-version", cluster.OpenshiftVersion)
```
Comment (eranco74) [inline internal/host/hostcommands/install_cmd.go:151]: The assisted-installer will default to 4.6 but things should work.
:wink:

File: skipper.yaml [hunk 4]
```diff
@@ -15,6 +15,9 @@ volumes:
   - $HOME/.kube/:$HOME/.kube
   - $HOME/.docker/:$HOME/.docker
   - $KUBECONFIG:$KUBECONFIG
+  - /etc/subuid:/etc/subuid
```
Comment (eranco74) [inline skipper.yaml:18]: Why?

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/host/hostcommands/install_cmd.go (hunk 3 above)
Comment (tsorya) [reply internal/host/hostcommands/install_cmd.go:151]: we just don't need it for day2 at all , it is not only if we have or not openshift version

**Pull request merged**